cmd/telemetrygen/internal/metrics/worker.go

```diff
@@ -28,29 +29,63 @@ type worker struct {
 
 func (w worker) simulateMetrics(res *resource.Resource, exporter sdkmetric.Exporter) {
 	limiter := rate.NewLimiter(w.limitPerSecond, 1)
-	var i int64
 
+	var i int64
 	for w.running.Load() {
-		rm := metricdata.ResourceMetrics{
-			Resource: res,
-			ScopeMetrics: []metricdata.ScopeMetrics{
-				{
-					Metrics: []metricdata.Metrics{
+		var metrics []metricdata.Metrics
+		if w.metricType == metricTypeGauge || w.metricType == metricTypeAll {
+			metrics = append(metrics, metricdata.Metrics{
+				Name: "gen.metric.gauge",
+				Data: metricdata.Gauge[int64]{
+					DataPoints: []metricdata.DataPoint[int64]{
+						{
+							StartTime: time.Now(),
+							Time:      time.Now().Add(1 * time.Second),
```

Comment:
Let's leave this as it was, there is no need to change the gauge on this PR
```suggestion
							Time:      time.Now(),
```